### DISCUSSION TITLE
Where does Pacman download from?

### DISCUSSION THREAD

**Discussion by denivic** [msys2/MINGW-packages#18843] [Q&A] [answered]
I need to download the GTK3 package for Windows server 2022 (`pacman -S mingw-w64-x86_64-gtk3`) but it won't download anything at all. I suspect it is due to some firewall rules, which is why I would like to know which ports or adresses I need to open up for to make this work.

**Comment by Biswa96** [comment, marked as answer]
pacman just follows the mirrorlist files in /etc/pacman.d directory. More info here https://wiki.archlinux.org/title/Mirrors

**Reply by denivic** [reply]
Thank you for the reply, appreciate it. :)

I have since my post managed to find that list and whitelisted only the primary ones to begin with through the firewall. I am being a little cautious as some of these mirrors are from China and Russia. This in itself is not an issue, but something to note if you are working with sensitive personal data.

Do you know why Tencent is one of the links in the mirrorlist?

**Reply by Biswa96** [reply]
> Do you know why Tencent is one of the links in the mirrorlist?

Anyone can host a pacman repository mirror.

**Reply by Biswa96** [reply]
> ...whitelisted only the primary ones to begin with through the firewall.

You can also remove or comment out the mirrors in those mirrorlist files. Make sure to repeat same changes with all the mirrorlist files. That ArchLinux wiki page has more information.
